File: lib/utils/aes_encryption.dart
import 'package:encrypt/encrypt.dart';

class AESEncryption {
  // Claves proporcionadas por el sistema
  static const String _keyString = '92EQ11A79WOP3B9I';  // 16 caracteres para AES-128
  static const String _ivString = 'SSL23LALLDL14378';   // 16 caracteres para IV
  
  static final _key = Key.fromBase16(_stringToHex(_keyString));
  static final _iv = IV.fromBase16(_stringToHex(_ivString));
  static final _encrypter = Encrypter(AES(_key, mode: AESMode.cbc, padding: 'PKCS7'));

  /// Convierte string a hexadecimal
  static String _stringToHex(String input) {
    return input.codeUnits.map((unit) => unit.toRadixString(16).padLeft(2, '0')).join('');
  }

  /// Encripta un texto plano
  static String encrypt(String plainText) {
    try {
      final encrypted = _encrypter.encrypt(plainText, iv: _iv);
      return encrypted.base64;
    } catch (e) {
      throw Exception('Error al encriptar: $e');
    }
  }

  /// Desencripta un texto encriptado (en base64)
  static String decrypt(String encryptedText) {
    try {
      final encrypted = Encrypted.fromBase64(encryptedText);
      final decrypted = _encrypter.decrypt(encrypted, iv: _iv);
      return decrypted;
    } catch (e) {
      throw Exception('Error al desencriptar: $e');
    }
  }

  /// Verifica si una contraseña plana coincide con una encriptada
  static bool verifyPassword(String plainPassword, String encryptedPassword) {
    try {
      final encryptedPlain = encrypt(plainPassword);
      return encryptedPlain == encryptedPassword;
    } catch (e) {
      return false;
    }
  }

  /// Método de prueba para validar la encriptación
  static void testEncryption() {
    const testPassword = 'test123';
    
    try {
      // Encriptar
      final encrypted = encrypt(testPassword);
      print('🔒 Contraseña original: $testPassword');
      print('🔐 Contraseña encriptada: $encrypted');
      
      // Desencriptar
      final decrypted = decrypt(encrypted);
      print('🔓 Contraseña desencriptada: $decrypted');
      
      // Verificar
      final isValid = verifyPassword(testPassword, encrypted);
      print('✅ Verificación: ${isValid ? 'EXITOSA' : 'FALLIDA'}');
      
    } catch (e) {
      print('❌ Error en prueba: $e');
    }
  }
}

File: backend_new/src/main/java/com/lytiks/backend/util/AESEncryption.java
package com.lytiks.backend.util;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;

public class AESEncryption {
    
    // Claves proporcionadas por el sistema
    private static final String KEY_STRING = "92EQ11A79WOP3B9I";  // 16 caracteres para AES-128
    private static final String IV_STRING = "SSL23LALLDL14378";   // 16 caracteres para IV
    
    private static final String ALGORITHM = "AES";
    private static final String TRANSFORMATION = "AES/CBC/PKCS5Padding";
    
    /**
     * Encripta un texto plano usando AES
     */
    public static String encrypt(String plainText) {
        try {
            SecretKeySpec keySpec = new SecretKeySpec(KEY_STRING.getBytes("UTF-8"), ALGORITHM);
            IvParameterSpec ivSpec = new IvParameterSpec(IV_STRING.getBytes("UTF-8"));
            
            Cipher cipher = Cipher.getInstance(TRANSFORMATION);
            cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);
            
            byte[] encrypted = cipher.doFinal(plainText.getBytes("UTF-8"));
            return Base64.getEncoder().encodeToString(encrypted);
            
        } catch (Exception e) {
            throw new RuntimeException("Error al encriptar: " + e.getMessage(), e);
        }
    }
    
    /**
     * Desencripta un texto encriptado (en base64) usando AES
     */
    public static String decrypt(String encryptedText) {
        try {
            SecretKeySpec keySpec = new SecretKeySpec(KEY_STRING.getBytes("UTF-8"), ALGORITHM);
            IvParameterSpec ivSpec = new IvParameterSpec(IV_STRING.getBytes("UTF-8"));
            
            Cipher cipher = Cipher.getInstance(TRANSFORMATION);
            cipher.init(Cipher.DECRYPT_MODE, keySpec, ivSpec);
            
            byte[] encrypted = Base64.getDecoder().decode(encryptedText);
            byte[] decrypted = cipher.doFinal(encrypted);
            
            return new String(decrypted, "UTF-8");
            
        } catch (Exception e) {
            throw new RuntimeException("Error al desencriptar: " + e.getMessage(), e);
        }
    }
    
    /**
     * Verifica si una contraseña plana coincide con una encriptada
     */
    public static boolean verifyPassword(String plainPassword, String encryptedPassword) {
        try {
            String encryptedPlain = encrypt(plainPassword);
            return encryptedPlain.equals(encryptedPassword);
        } catch (Exception e) {
            return false;
        }
    }
    
    /**
     * Método de prueba para validar la encriptación
     */
    public static void testEncryption() {
        String testPassword = "test123";
        
        try {
            // Encriptar
            String encrypted = encrypt(testPassword);
            System.out.println("🔒 Contraseña original: " + testPassword);
            System.out.println("🔐 Contraseña encriptada: " + encrypted);
            
            // Desencriptar
            String decrypted = decrypt(encrypted);
            System.out.println("🔓 Contraseña desencriptada: " + decrypted);
            
            // Verificar
            boolean isValid = verifyPassword(testPassword, encrypted);
            System.out.println("✅ Verificación: " + (isValid ? "EXITOSA" : "FALLIDA"));
            
        } catch (Exception e) {
            System.out.println("❌ Error en prueba: " + e.getMessage());
        }
    }
}